plugin-build/src/main/kotlin/io/sentry/android/gradle/SentryPlugin.kt

```diff
@@ -36,84 +36,116 @@ class SentryPlugin : Plugin<Project> {
-                    val cliExecutable = getSentryCliPath(project)
-                    val sep = File.separator
+                    mappingFile = getMappingFile(project, variant)
+                } else {
+                    project.logger.info("[sentry] isMinifyEnabled is disabled.")
+                }
 
-                    // Setup the task to generate a UUID asset file
-                    val generateUuidTask = project.tasks.create(
-                        "generateSentryProguardUuid$taskSuffix",
-                        SentryGenerateProguardUuidTask::class.java
-                    ) {
-                        it.outputDirectory.set(
-                            project.file(
-                                "build${sep}generated${sep}assets${sep}sentry${sep}${variant.name}"
+                variant.outputs.all { variantOutput ->
+                    val taskSuffix = varNameCapitalize + variantOutput.name.capitalize(Locale.ROOT)
+
+                    if (isMinifyEnabled) {
+                        // Setup the task to generate a UUID asset file
+                        val generateUuidTask = project.tasks.register(
+                            "generateSentryProguardUuid$taskSuffix",
+                            SentryGenerateProguardUuidTask::class.java
+                        ) {
+                            it.outputDirectory.set(
+                                project.file(
+                                    "build${sep}generated${sep}assets${sep}sentry${sep}${variant.name}"
+                                )
                             )
+                        }
+                        variant.mergeAssetsProvider.configure { it.dependsOn(generateUuidTask) }
+
+                        // Setup the task that uploads the proguard mapping and UUIDs
+                        val uploadSentryProguardMappingsTask = project.tasks.register(
+                            "uploadSentryProguardMappings$taskSuffix",
+                            SentryUploadProguardMappingsTask::class.java
+                        ) {
+                            it.dependsOn(generateUuidTask)
+                            it.workingDir(project.rootDir)
+                            it.cliExecutable.set(cliExecutable)
+                            it.sentryProperties.set(
+                                sentryProperties?.let { file -> project.file(file) }
+                            )
+                            it.mappingsUuid.set(generateUuidTask.get().outputUuid)
+                            mappingFile?.let { mapFile ->
+                                it.mappingsFile.set(mapFile)
+                            }
+                            it.autoUpload.set(extension.autoUpload.get())
+                            it.sentryOrganization.set(sentryOrgParameter)
+                            it.sentryProject.set(sentryProjectParameter)
+                        }
+                        variant.register(uploadSentryProguardMappingsTask.get())
```

Comment:
Doing this will eagerly create the `uploadSentryProguardMappingsTask` task.
I am wondering if we need `variant.register` here since we already configure tasks dependency below.

Reply:
This one will go away as it was removed in prior PR 👌